src/v/coproc/router.cc

```diff
@@ -24,8 +24,12 @@
 #include <seastar/core/gate.hh>
 #include <seastar/core/loop.hh>
```

Comment:
There is a lot of complex stuff happening in this patch. The commit message needs to explain what processes are interacting together through the promise/future's that have been added and are maintained in new indexes. It isn't enough to read this code and get an understanding of the potential concurrency and lifetime issues.

Reply:
gotcha, i'll write a more extensive commit msg

Reply:
thanks!